addon/components/select-2.js

```diff
@@ -411,6 +430,21 @@ var Select2Component = Ember.Component.extend({
     this.watchDisabled();
   },
 
+  createRecordFromTag: function(tag) {
+    var type;
+    if (this.get('value') instanceof Ember.ArrayProxy) {
+      type = this.get('value.content.type');
+    } else {
+      type = this.get('value.type');
+    }
+
+    var record = this.get('container').lookup('store:main').createRecord(type.typeKey);
+
+    record.set(this.get('optionLabelPath'), tag[this.get('optionLabelPath')]);
+
+    return record;
+  },
+
```

Comment:
This looks like a simple way of doing things, which is nice. If someone wants to have more control over creating records, we can still change this method to *optionally* use a factory method on the controller (similar to the `query` action). So fine with me, too.